src/main/kotlin/com/terning/server/kotlin/ui/api/ExceptionHandler.kt

```diff
@@ -0,0 +1,70 @@
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.HttpStatusCode
+import org.springframework.http.ResponseEntity
+import org.springframework.http.converter.HttpMessageNotReadableException
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.context.request.WebRequest
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
+
+@RestControllerAdvice
+class ExceptionHandler : ResponseEntityExceptionHandler() {
+    override fun handleHttpMessageNotReadable(
+        ex: HttpMessageNotReadableException,
+        headers: HttpHeaders,
+        status: HttpStatusCode,
+        request: WebRequest
+    ): ResponseEntity<Any> {
+        logger.error("message", ex)
+        val message = when (val exception = ex.cause) {
+            is MismatchedInputException -> "${exception.path.lastOrNull()?.fieldName ?: "UnknownField"}: 널이어서는 안됩니다"
+            is InvalidFormatException -> "${exception.path.lastOrNull()?.fieldName ?: "UnknownField"}: 올바른 형식이어야 합니다"
+            else -> exception?.message.orEmpty()
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+            .body(ApiResponse.error(message))
+    }
+
+    override fun handleMethodArgumentNotValid(
+        ex: MethodArgumentNotValidException,
+        headers: HttpHeaders,
+        status: HttpStatusCode,
+        request: WebRequest
+    ): ResponseEntity<Any> {
+        logger.error("message", ex)
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+            .body(ApiResponse.error(ex.messages()))
+    }
+
+    @ExceptionHandler(IllegalArgumentException::class, IllegalStateException::class)
+    fun handleBadRequestException(exception: RuntimeException): ResponseEntity<ApiResponse<Unit>> {
+        logger.error("message", exception)
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+            .body(ApiResponse.error(exception.message))
+    }
+
+    @ExceptionHandler(EntityNotFoundException::class)
+    fun handleNotFoundException(exception: EntityNotFoundException): ResponseEntity<ApiResponse<Unit>> {
+        logger.error("message", exception)
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+            .body(ApiResponse.error(exception.message))
+    }
+
+    @ExceptionHandler(Exception::class)
+    fun handleGlobalException(exception: Exception): ResponseEntity<ApiResponse<Unit>> {
+        logger.error("message", exception)
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+            .body(ApiResponse.error(exception.message))
+    }
+
+    private fun MethodArgumentNotValidException.messages(): String {
+        return bindingResult.fieldErrors.joinToString(", ") { "${it.field}: ${it.defaultMessage.orEmpty()}" }
+    }
```

Comment:
```suggestion
    private fun MethodArgumentNotValidException.messages(): String =
        bindingResult.fieldErrors.joinToString(", ") { "${it.field}: ${it.defaultMessage.orEmpty()}" }
```
저는 return을 표현식으로 해야만 하는 병이 있어요

Reply:
귀여운 병이네요 ㅎㅎ 저도 전염될까 봐 걱정이에요. 마스크라도 써야 할까 봐요 😷ㅎㅎ

유빈님께서 예전에 코틀린 피드백 주셨을 때, `return`보다는 표현식으로 작성하는 걸 권장해주셨던 게 인상 깊었어요. 그 이후로 코드를 작성할 때마다 표현식으로 바꿔보려는 시도를 하고 있는데요!

이번 경우엔 오히려 `return`을 명시적으로 사용하는 편이 더 낫지 않을까 하는 생각이 들었습니다.
첫 번째 이유는 가시성입니다. 표현식 형태로 작성하면 한 줄에 모든 로직이 담기다 보니, 라인이 가로로 길어지면서 오히려 읽기 어려워지는 경우가 있는 것 같아요. `return`으로 명확히 구분되면 구조적으로 더 눈에 잘 들어오고요.

두 번째는 유지보수성입니다. 결국 우리가 작성한 코드는 여러 사람이 함께 보고, 고치고, 확장해나가는 코드이기 때문에, 한눈에 구조가 파악되는 쪽이 장기적으로 봤을 때 유지보수에 더 유리하다고 느꼈어요.

물론 성능상 차이가 있고, 표현식 방식이 더 일관성 있게 보이거나 팀 스타일과 잘 맞는다면 적극적으로 도입하고 싶습니다. 다만 아직은 실용적인 면에서 `return`이 더 낫지 않나 싶어 이렇게 작성했는데, 혹시 이 부분에 대해 유빈님은 어떻게 생각하시는지 궁금해요! 😊

Reply:
제가 생각할 때 코틀린의 장점은 간결하다는 점인 것 같아요! 표현식을 사용할 수 있다는 점이 그 이유 중 하나이기도 하구요
그래서 표현식으로 나타낼 수 있는 경우는 적극적으로 사용해주는 게 '코틀린스러운' 방식이라고 생각했습니다!

성능상 차이는 미미하겠지만 그 언어의 장점을 살려서 작성하는 습관을 기르는 게 좋을 것 같아서 의견을 제시했던 겁니당..!!

Reply:
유빈님의 의견에 전적으로 공감해요! 🙌  
저는 사실 평소에는 `return`을 사용하는 스타일이었어요. 자바에서 오랫동안 그렇게 써오다 보니, 코틀린에서도 자연스럽게 `return`을 쓰게 되더라고요. 😅

근데 이번에 말씀 주신 걸 보면서, 표현식으로도 충분히 가독성 있게 표현할 수 있겠다는 생각이 들었어요. 특히 제가 착각하고 있었던 게, `=`을 쓰면 무조건 가로로 길게 작성해야 된다고 생각했던 건데, 줄바꿈을 적절히 활용하면 오히려 더 깔끔하게 표현할 수 있다는 걸 알게 됐어요. 덕분에 코틀린의 장점을 더 잘 살릴 수 있겠다는 생각이 들었습니다!

앞으로도 습관처럼 `return`이 나올 수 있긴 하겠지만 😂 그런 부분들 편하게 피드백 주시면 저도 계속해서 더 코틀린스러운 방식으로 고쳐가보겠습니다! 피드백 진심으로 감사드려요 💙